### PR TITLE
Internal functions for KMS

### PR DESCRIPTION
**Pull request by akhilmhdh**
# Description 📣

1. Implemented functions for KMS
2. Includes - Root key generator, Generate Symmetric KMS key, Encrypt and decrypt via KMS key

## Type ✨

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

**Comment by dangtony98** [comment]
@akhilmhdh Seems like backend tests are failing at the moment.

File: backend/src/services/kms/kms-service.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      });
+    }
+
+    // check if KMS root key was already generated and saved in DB
+    const kmsRootConfig = await kmsRootConfigDAL.findById(KMS_ROOT_CONFIG_UUID);
+    const cipher = symmetricCipherService(SymmetricEncryption.AES_GCM_256);
+    if (kmsRootConfig) {
+      if (lock) await lock.release();
+      logger.info("KMS: Encrypted ROOT Key found from DB. Decrypting.");
+      const decryptedRootKey = cipher.decrypt(kmsRootConfig.encryptedRootKey, Buffer.from(encryptionKey, "utf8"));
```
Comment (dangtony98) [inline backend/src/services/kms/kms-service.ts:93]: This line fails when `ROOT_ENCRYPTION_KEY` is used as a `base64` encoded string.

File: backend/src/services/kms/kms-service.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      // set the flag so that other instancen nodes can start
+      await keyStore.setItemWithExpiry(KMS_ROOT_CREATION_WAIT_KEY, KMS_ROOT_CREATION_WAIT_TIME, "true");
+      logger.info("KMS: Loading ROOT Key into Memory.");
+      ROOT_ENCRYPTION_KEY = decryptedRootKey;
+      return;
+    }
+
+    logger.info("KMS: Generating ROOT Key");
+    const newRootKey = randomSecureBytes(32);
+    const encryptedRootKey = cipher.encrypt(newRootKey, Buffer.from(encryptionKey, "utf8"));
```
Comment (dangtony98) [inline backend/src/services/kms/kms-service.ts:103]: This line fails when `ROOT_ENCRYPTION_KEY` is used as a `base64` encoded string.

**Review by dangtony98 — APPROVED** [review]

File: .github/workflows/check-api-for-breaking-changes.yml [hunk 3]
```diff
@@ -40,13 +40,14 @@ jobs:
...
           JWT_AUTH_SECRET: something-random
+          ENCRYPTION_KEY: 4bnfe4e407b8921c104518903515b218
       - uses: actions/setup-go@v5
         with:
           go-version: '1.21.5'
       - name: Wait for container to be stable and check logs
         run: |
           SECONDS=0
-          HEALTHY=0
+ r        HEALTHY=0
```
Comment (Apetree100122) [inline .github/workflows/check-api-for-breaking-changes.yml:50]: why is there an r
on line 50 and no ; after the 0 would not that cause error silently unnoticed iF you can please review multiple additions that i seen unnessacary.

**Review by Apetree100122 — COMMENTED** [review]
please review suggestions and see if helps

**Pull request merged**